feature/gnmi/otg_tests/telemetry_port_speed_test/telemetry_port_speed_test.go

```diff
@@ -0,0 +1,455 @@
+				gnmi.BatchDelete(resetBatch, gnmi.OC().NetworkInstance(deviations.DefaultNetworkInstance(tc.dut)).Interface(tc.aggID+".0").Config())
+			}
+			gnmi.BatchDelete(resetBatch, aggPath.Config())
+			gnmi.BatchDelete(resetBatch, d.Lacp().Interface(tc.aggID).Config())
+			resetBatch.Set(t, tc.dut)
+		}
+		gnmi.Delete(t, tc.dut, aggPath.Config())
+	})
+}
+
+func (tc *testCase) configureATE(t *testing.T) {
+	if len(tc.atePorts) < 2 {
+		t.Fatalf("Testbed requires at least 2 ports, got: %v", tc.atePorts)
+	}
+	agg := tc.top.Lags().Add().SetName("lag")
+	if tc.lagType == lagTypeSTATIC {
+		lagId, _ := strconv.Atoi(tc.aggID)
+		agg.Protocol().SetChoice("static").Static().SetLagId(uint32(lagId))
+		for i, p := range tc.atePorts[0:] {
+			port := tc.top.Ports().Add().SetName(p.ID())
+			newMac, err := incrementMAC(ateIPs.MAC, i+1)
+			if err != nil {
+				t.Fatal(err)
+			}
+			agg.Ports().Add().SetPortName(port.Name()).Ethernet().SetMac(newMac).SetName("LAGRx-" + strconv.Itoa(i))
+		}
+	} else {
+		agg.Protocol().SetChoice("lacp")
+		agg.Protocol().Lacp().SetActorKey(1).SetActorSystemPriority(1).SetActorSystemId(ateIPs.MAC)
+		for i, p := range tc.atePorts[0:] {
+			port := tc.top.Ports().Add().SetName(p.ID())
+			newMac, err := incrementMAC(ateIPs.MAC, i+1)
+			if err != nil {
+				t.Fatal(err)
+			}
+			lagPort := agg.Ports().Add().SetPortName(port.Name())
+			lagPort.Ethernet().SetMac(newMac).SetName("LAGRx-" + strconv.Itoa(i))
+			lagPort.Lacp().SetActorActivity("active").SetActorPortNumber(uint32(i) + 1).SetActorPortPriority(1).SetLacpduTimeout(0)
+		}
+	}
+
+	// Disable FEC for 100G-FR ports because Novus does not support it.
+	p100gbasefr := []string{}
+	for _, p := range tc.atePorts {
+		if p.PMD() == ondatra.PMD100GBASEFR {
+			p100gbasefr = append(p100gbasefr, p.ID())
+		}
+	}
+
+	if len(p100gbasefr) > 0 {
+		l1Settings := tc.top.Layer1().Add().SetName("L1").SetPortNames(p100gbasefr)
+		l1Settings.SetAutoNegotiate(true).SetIeeeMediaDefaults(false).SetSpeed("speed_100_gbps")
+		autoNegotiate := l1Settings.AutoNegotiation()
+		autoNegotiate.SetRsFec(false)
+	}
+
+	dstDev := tc.top.Devices().Add().SetName(agg.Name() + ".dev")
+	dstEth := dstDev.Ethernets().Add().SetName(ateIPs.Name + ".Eth").SetMac(ateIPs.MAC)
+	dstEth.Connection().SetLagName(agg.Name())
+	dstEth.Ipv4Addresses().Add().SetName(ateIPs.Name + ".IPv4").SetAddress(ateIPs.IPv4).SetGateway(dutIPs.IPv4).SetPrefix(uint32(ateIPs.IPv4Len))
+	dstEth.Ipv6Addresses().Add().SetName(ateIPs.Name + ".IPv6").SetAddress(ateIPs.IPv6).SetGateway(dutIPs.IPv6).SetPrefix(uint32(ateIPs.IPv6Len))
+
+	tc.ate.OTG().PushConfig(t, tc.top)
+	tc.ate.OTG().StartProtocols(t)
+
+}
+
+func (tc *testCase) verifyDUT(t *testing.T, numPort int) {
+	dutPort := tc.dut.Port(t, "port1")
+	want := int(dutPort.Speed()) * numPort * 1000
+	val, _ := gnmi.Watch(t, tc.dut, gnmi.OC().Interface(tc.aggID).Aggregation().LagSpeed().State(), 60*time.Second, func(val *ygnmi.Value[uint32]) bool { return val.IsPresent() }).Await(t)
```

Comment:
Should check for the value here instead of `IsPresent`. We've gotten some flakiness here for the OTG version of the test, I suspect because of the swap in ordering between ATE/DUT config.

Reply:
Hopefully this change is what was needed.